wolfssl/wolfcrypt/random.h

```diff
@@ -39,18 +39,22 @@
 #define RNG_MAX_BLOCK_LEN (0x10000)
 
 #ifndef HAVE_FIPS /* avoid redefining structs and macros */
+
 #if defined(WOLFSSL_FORCE_RC4_DRBG) && defined(NO_RC4)
     #error Cannot have WOLFSSL_FORCE_RC4_DRBG and NO_RC4 defined.
 #endif /* WOLFSSL_FORCE_RC4_DRBG && NO_RC4 */
-#if defined(HAVE_HASHDRBG) || defined(NO_RC4)
-    #ifdef NO_SHA256
-        #error "Hash DRBG requires SHA-256."
-    #endif /* NO_SHA256 */
 
-    #include <wolfssl/wolfcrypt/sha256.h>
-#else /* HAVE_HASHDRBG || NO_RC4 */
-    #include <wolfssl/wolfcrypt/arc4.h>
-#endif /* HAVE_HASHDRBG || NO_RC4 */
+#if defined(HAVE_HASHDRBG)
+    #ifdef NO_RC4
```

Comment:
This doesn't look right.  What if RC4 is on?  Then the SHA256 check doesn't happen.